pkg/worker/verifier/producer_worker.go

```diff
@@ -346,7 +357,15 @@ func (pw *ProducerWorker) produceInner(n int64) (int64, []BadOffset, error) {
 		sentAt := time.Now()
 		handler := func(r *kgo.Record, err error) {
 			concurrent.Release(1)
-			util.Chk(err, "Produce failed: %v", err)
+
+			if err != nil {
+				pw.Status.OnFail()
+				errHandler := util.Die
+				if pw.tolerateFailedProduce {
+					errHandler = log.Warnf
+				}
+				errHandler("Produce failed: %v", err)
```

Comment:
Looks fine-ish but I'm not sure it is safe to continue producing from there in the general case because then we'll get in a case where `expectOffset != r.Offset` will be true thus erroring the producer and it dying anyway at https://github.com/redpanda-data/kgo-verifier/blob/99de2ad6a9cf0320ec1d60d73b1ae66581d4cade/cmd/kgo-verifier/main.go#L256

Wouldn't it be better to just exit early from here somehow but leaving the status handler alive instead of continuing retrying?

Reply:
For my use-case it'll be nice if it carries on banging into the locked door. Do you think it'll be okay if I tried to make it restart the producer and continue?